Windows GHA CI green on invalid opam file

Spotted on #29486:

When encountering an invalid opam file, the Windows GHA workflows print a warning as expected,
but still complete with a green light. 


https://github.com/ocaml/opam-repository/actions/runs/22796601588/job/66132129043?pr=29486
```
Run $failed = $false

<><> Updating package repositories ><><><><><><><><><><><><><><><><><><><><><><>
[default] synchronised from file://D:/a/opam-repository/opam-repository
Warning:  Could not read file D:\opamroot\repo\default\packages\biotk\biotk.0.4\opam. skipping:
          At D:\opamroot\repo\default\packages\biotk\biotk.0.4\opam:24:13-24:14::
          '.' is not a valid token
Now run 'opam upgrade' to apply any package updates.
Testing biotk.0.4
  [ERROR] Package biotk has no version 0.4.
biotk.0.4 is not installable. Skip.
```

I think we just need to check the exit code of the `opam update` too.

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -81,9 +81,14 @@ jobs:
           }
           Foreach ($pkg in '${{ matrix.packages }}' -split ' ') {
             Write-Host "::group::Testing `e[1;34m$pkg`e[0m"
+            $pkg_dir = $pkg.Split('.')[0]
+            opam lint -v "packages/$pkg_dir/$pkg"
+            if ($LASTEXITCODE -ne 0) {
+              Write-Host "$pkg failed linting."; $failed = $true
+            }
             opam install -v --color=always --confirm-level=unsafe-yes "$pkg"
             $install_exit_code = $LASTEXITCODE
-            opam config list $pkg.Split('.')[0] | Skip-Until ':opamfile'
+            opam config list $pkg_dir | Skip-Until ':opamfile'
             Write-Host "::endgroup::"
             switch ($install_exit_code) {
               0 { Write-Host "`e[1;32m$pkg installed successfully`e[0m."; Break }
@@ -95,5 +100,5 @@ jobs:
             Write-Host
           }
           if ($failed) {
-            throw "One or more packages failed to build"
+            throw "One or more packages failed to lint or build"
           }

